src/apis/ProxyController.php

```diff
@@ -29,13 +30,26 @@
  */
 class ProxyController extends Controller
 {
+    /**
+     * @var Connection
+     * @since 2.0.0
+     */
+    protected $db;
+
     /**
      * @var array A list of tables which will be ignored and can not be synced with the proxy command.
      */
     protected $ignoreTables = [
         'migration', 'admin_proxy_build', 'admin_proxy_machine',
     ];
-    
+
+    public function beforeAction($action)
+    {
+        $this->db = Yii::$app->get($this->module->proxyConnectionName);
```

Comment:
i think we should use Instance::ensure for this use case: example from the yii docs:

$this->db = Instance::ensure($this->db, 'yii\db\Connection');